circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -92,20 +92,52 @@ trait CirceInstances extends JawnInstances {
   implicit def jsonEncoder[F[_]: Applicative]: EntityEncoder[F, Json] =
     jsonEncoderWithPrinter(defaultPrinter)
 
-  def jsonEncoderWithPrinter[F[_]: Applicative](printer: Printer): EntityEncoder[F, Json] =
+  private def fromJsonToChunk(printer: Printer, cs: Charset)(json: Json): Chunk[Byte] =
+    Chunk.byteBuffer(printer.printToByteBuffer(json, cs.nioCharset))
+
+  def jsonEncoderWithPrinter[F[_]: Applicative](
+      printer: Printer,
+      cs: Charset = DefaultCharset): EntityEncoder[F, Json] =
     EntityEncoder[F, Chunk[Byte]]
-      .contramap[Json] { json =>
-        val bytes = printer.printToByteBuffer(json)
-        Chunk.byteBuffer(bytes)
-      }
+      .contramap[Json](fromJsonToChunk(printer, cs))
       .withContentType(`Content-Type`(MediaType.application.json))
 
   def jsonEncoderOf[F[_]: Applicative, A](implicit encoder: Encoder[A]): EntityEncoder[F, A] =
     jsonEncoderWithPrinterOf(defaultPrinter)
 
-  def jsonEncoderWithPrinterOf[F[_]: Applicative, A](printer: Printer)(
-      implicit encoder: Encoder[A]): EntityEncoder[F, A] =
-    jsonEncoderWithPrinter[F](printer).contramap[A](encoder.apply)
+  def jsonEncoderWithPrinterOf[F[_]: Applicative, A](
+      printer: Printer,
+      cs: Charset = DefaultCharset)(implicit encoder: Encoder[A]): EntityEncoder[F, A] =
+    jsonEncoderWithPrinter[F](printer, cs).contramap[A](encoder.apply)
+
+  implicit def streamJsonArrayEncoder[F[_]: Applicative]: EntityEncoder[F, Stream[F, Json]] =
+    streamJsonArrayEncoderWithPrinter(defaultPrinter)
+
+  /** An [[EntityEncoder]] for a [[Stream]] of JSONs, which will encode it as a single JSON array. */
+  def streamJsonArrayEncoderWithPrinter[F[_]: Applicative](
+      printer: Printer,
+      cs: Charset = DefaultCharset): EntityEncoder[F, Stream[F, Json]] = {
+    def fromStr(str: String): Chunk[Byte] =
+      Chunk.byteBuffer(cs.nioCharset.encode(str))
```

Comment:
Might be able to micro-optimize this by having constant chunks for the delimiters, but I doubt it makes a huge difference.  It would be tractable if the `cs` domain were reduced to three, or one.

Reply:
Given I will be removing the charset and just leave the default one, I can do this.